packages/vats/test/test-provisionPool.js

```diff
@@ -350,8 +371,124 @@ test('makeBridgeProvisionTool handles duplicate requests', async t => {
     'depositFacet lookup finds the same object',
   );
   t.deepEqual(
-    await E(p1).getCurrentAmount(),
+    await E(purse).getCurrentAmount(),
     fees.make(500n),
     'received more starter funds',
   );
 });
+
+test('provisionPool revives old wallets', async t => {
+  const { zoe, installs, storageRoot, committeeCreator } = t.context;
+
+  // make a mock wallet factory and setup its bank
+  const oldAddr = 'addr_old';
+  const newAddr = 'addr_new';
+  const { walletFactory, purses, fees, bankManager } =
+    await makeWalletFactoryKitForAddresses([oldAddr, newAddr]);
+  const oldPurse = /** @type {Purse} */ (purses.get(oldAddr));
+  const newPurse = /** @type {Purse} */ (purses.get(newAddr));
+  const poolBank = await E(bankManager).getBankForAddress('pool');
+  const poolPurse = E(poolBank).getPurse(fees.brand);
+  await E(poolPurse).deposit(
+    await E(fees.mint).mintPayment(fees.make(scale6(100))),
+  );
+
+  // start a provisionPool contract with mock governance terms
+  const starterAmount = fees.make(250n);
+  const initialPoserInvitation = await E(committeeCreator).getPoserInvitation();
```

Comment:
After writing this boilerplate for starting a governed contract 1 too many times, we came up with a puppet governance test tool that might be handy here.

https://github.com/Agoric/agoric-sdk/blob/master/packages/governance/tools/puppetGovernance.js#L1

Reply:
I'm working with existing test context here and am not confident I understand enough to adopt that utility (but we could pair on that if you think it's worthwhile).